### PR TITLE
Add API on FocusProvider to set focus point

### PR DESCRIPTION
**Pull request by kircher1**
## Overview
Adds an API on FocusProvider to override the focus point for a given pointer.

## Changes
- Fixes: #5015

### CODE REVIEW

**Comment by julenka** [comment]
Hi @kircher1 just checking in on this PR -- anything you need from me or any other reviewers to unblock you? I made a few suggestions for changes but after that I think it's ready to be updated from draft.

**Comment by kircher1** [comment]
Thanks for the feedback @julenka! All good ideas.

I'm not actively looking at this currently. If someone wants to implement this ASAP, feel free, but please coordinate here so we don't duplicate efforts. Otherwise, I should have time soon (probably next week) to wrap this up. Will reply back when I start working on it.

**Comment by julenka** [comment]
Next week sounds great, thanks for clarifying! I was just checking in, we don't have anybody else currently working on this. Plenty of other bugs to fix :-)

**Comment by kircher1** [comment]
Alright, I'm working on the suggestions now: Pulling the API up to the interface and verifying the effects on other FocusProvider related APIs.

**Comment by kircher1** [comment]
@julenka, the latest commit will bring this API to the focus provider interface. Tests have bee added to verify the other points as well!

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs [hunk 1]
```diff
@@ -91,6 +91,44 @@ public IEnumerator TestGazeProviderTargetNotNull()
 
             Assert.NotNull(CoreServices.InputSystem.GazeProvider.GazeTarget, "GazeProvider target is null when looking at an object with hand raised");
         }
+
+        /// <summary>
+        /// Ensure that the gaze provider hit result is not null when looking at an object,
```
Comment (julenka) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs]: Can you fix up the comment please?

File: Assets/MixedRealityToolkit/Interfaces/InputSystem/IMixedRealityFocusProvider.cs [hunk 2]
```diff
@@ -54,6 +54,15 @@ public interface IMixedRealityFocusProvider : IMixedRealityService, IMixedRealit
         /// <param name="focusDetails"></param>
         bool TryGetFocusDetails(IMixedRealityPointer pointer, out FocusDetails focusDetails);
 
+        /// <summary>
+        /// Sets the FocusDetails of the specified pointer, overriding the focus point that was currently set. This can be used to change
+        /// the FocusDetails of a specific pointer even if focus is locked.
```
Comment (julenka) [inline Assets/MixedRealityToolkit/Interfaces/InputSystem/IMixedRealityFocusProvider.cs]: Could you provide an example of why / when you might want to do this (for example when panning content)?

**Comment by julenka** [comment]
Okay, looks like there may be a better way to get cursor to not stick to `HandInteractionPanZoom` and UnityUI canvases, using `IsPositionLockedOnFocusLock`, based on comments in #5836. So, I'll go ahead and complete this PR without those changes.

**Pull request merged**